src/vs/base/common/winjs.base.d.ts

```diff
@@ -134,5 +134,5 @@ export declare class PPromise<C, P> extends TPromise<C> {
 	public static join<C, P>(promises: PPromise<C, P>[]): PPromise<C, P[]>;
 	public static join<C, P>(promises: { [n: string]: PPromise<C, P> }): PPromise<{ [n: string]: C }, P>;
 	public static any<C, P>(promises: PPromise<C, P>[]): PPromise<{ key: string; value: PPromise<C, P>; }, P>;
-	public static wrapError<V>(error: any): TPromise<V>;
+	public static wrapError(error: Error): TPromise<never>;
```

Comment:
I know, I have uncommitted changes locally :)

>  Especially, when returning a rejected-promise from a function that returns Promise<T> isn't possible with the never-approch.

Could you clarify what you mean by this? One of the traits of the `never` type is that you can always assign/return it, e.g. a function can have a return type `Promise<number>` but you can return `Promise<never>`. Just like a function can return `number`, but actually throw an Error in some cases (which means its real return type is `number | never`, which is equivalent to `number`, because `never` always disappears in union types).

lib.d.ts actually will return `Promise<never>` if you don't pass a type argument: https://sourcegraph.com/github.com/Microsoft/TypeScript@master/-/blob/lib/lib.es2015.promise.d.ts?q=reject#L200-201
I am pretty sure they only left the old version for backwards-compatibility as the `never` type is quite new